general/our-software.md

```diff
@@ -0,0 +1,98 @@
+
+Regardless, projects are only "good" to work on if they're done to a workable standard. This is sometimes hard to achieve
+in a longstanding project that doesn't have the time or budget to be rewritten or ripped apart.
+
+One way you can ensure the slow improvement of a codebase is to **leave a file better than you found it**. Sometimes it's
+easy to shift the blame of technical debt onto others than to get stuck in and sort issues out, but adding to a poor
+codebase isn't going to do anyone any good.
+
+### Commercial Viability
+Now to contrast on everything that we just spoke about. Business is business and sometimes there are time constraints to
+how *nice* we can make the UI or what *"helpful features"* we can put in.
+
+#### MoSCoW Method
+Shamelessly pulled from [wikipedia](https://en.wikipedia.org/wiki/MoSCoW_method);
+
+> The MoSCoW method is a prioritization technique used in management, business analysis, project management, and software 
+development to reach a common understanding with stakeholders on the importance they place on the delivery of each 
+requirement; it is also known as MoSCoW prioritization or MoSCoW analysis.
+
+This becomes vital when planning the development schedule of a project. Knowing what the customer needs over wants quickly
+becomes important when it comes to stretching their budget as far as it will go. They might *want* a rear view camera
+in their car, but they should probably have the engine first.
+
+#### Make it quickly
+There is a certain art to building software to a budget, and building solid software at that. Often developers can find
+that they want to move onto the less fundamental and less boring part of a project to focus on some interesting logic, 
+or that they want to see a view scaffolded completely with the must-have tasks **and** the nice-to-haves.
+
+No matter what you do - **PLEASE, PLEASE** refrain from doing this. There are times when you will be building some repetitive
+CRUD functionality or building some static views that don't have much going on, but most of the time these are the most
+important to the user journey/experience. This ensures that the user will have the most complete *flow* at all times
+throughout the project
+
+##### Minimum Viable Product (MVP)
+We see this work more commonly in an MVP methodology, where you write the least amount of code
+to satisfy a story point. This enables us to get the most functionality in the smallest amount of time, push that
+functionality out to the end users for review and move on, knowing whether it aligns with the users thoughts or not.
+
+This also means that we get any nasty surprises or hiccups as soon as possible throughout the timeline, which helps
+project managers understand where time might be saved, or where we may need some more resource to bring in the deadline.
+
+This is usually explained best in a graphic.
+
+![Planned Timeline](images/our-software/original-timeline.png)
```

Comment:
It might be worth adding a small key to make it completely obvious what the different colours represent.